scrapy/core/downloader/middleware.py

```diff
@@ -38,23 +38,23 @@ def process_request(request):
                     raise _InvalidOutput('Middleware %s.process_request must return None, Response or Request, got %s' % \
                                          (method.__self__.__class__.__name__, response.__class__.__name__))
                 if response:
-                    defer.returnValue(response)
-            defer.returnValue((yield download_func(request=request, spider=spider)))
+                    return response
+            yield download_func(request=request, spider=spider)
```

Comment:
The value is not returned here anymore

Reply:
@wRAR should `return` be used here?
As the function is ending there, there is no need to "save" the state. which yield is useful for

Reply:
Not sure what do you mean by "saving the state" but the function must return the result of that `yield`, just like it did before. `yield` here is used to get the result of `download_func` (because of `@defer.inlineCallbacks`).

Reply:
@wRAR okay :+1: 

I'm still a little bit confused by  `The value is not returned here anymore` 
What exactly is it not returning ?

Reply:
It is not returning the result of `download_func`.

Reply:
(which was previously returned via `defer.returnValue` as you can see)

Reply:
Okay :+1:  I'm thinking of picking up this issue if the original author is not working on it.